pkg/state/cluster.go

```diff
@@ -21,19 +21,26 @@ import (
 	"sync"
 
 	"github.com/Masterminds/semver/v3"
+	apiserverconfigv1 "k8s.io/apiserver/pkg/apis/config/v1"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 
 	"k8c.io/kubeone/pkg/apis/kubeone"
 )
 
 type Cluster struct {
-	ControlPlane    []Host
-	StaticWorkers   []Host
-	ExpectedVersion *semver.Version
-	Lock            sync.Mutex
+	ControlPlane            []Host
+	StaticWorkers           []Host
+	ExpectedVersion         *semver.Version
+	EncryptionConfiguration *EncryptionConfiguration
+	Lock                    sync.Mutex
 }
 
+type EncryptionConfiguration struct {
+	Enable bool
+	Config *apiserverconfigv1.EncryptionConfiguration
+	Custom bool
```

Comment:
Unfortunately, yes. This is used in the `LiveCluster` object and is populated based on the actual configuration from the Control Plane. Since I can't tell if this configuration is custom or not, I had to use the file name to tell if this is custom config or not: https://github.com/kubermatic/kubeone/pull/1241/files#diff-3f3e2e23e35f00b4f97b18de406b4a5176ee7cdadaa8eb0d35ed424024dfcb06R474